### PR TITLE
Added string.explode function and tests

### LINKED ISSUE #34: bug(?) in string.split
Think I found an bug in the string.split function. If you use a multi-character separator but all the characters of the separator are the same it is treated as a single-character separator.

```lua
local a = string.split("Test the bees knees", "ee")
--Expected {"Test the b", "s kn", "s"}
--Got {"T", "st th", " b", "s kn", "s"}
```

[edit]
Any multi-character separator that start and/or ends with white space behave badly too. :/

Spent way longer than I intended trying to make string.gsub place nice in string.split. I'm just gonna push the php-like explode function from http://lua-users.org/wiki/SplitJoin I've been using for a few weeks now.

### PR DESCRIPTION
**Pull request by credomane**
Fixes #34

### CODE REVIEW

File: stdlib/string.lua [hunk 1]
```diff
@@ -52,3 +52,41 @@ function string.split(s, sep, pattern)
...
     return fields
 end
+
+--- Splits a string by another string
+-- @param s the string to split
+-- @param sep the boundary string
+-- @param limit If limit is set and positive, the returned array will contain a maximum of limit elements with the last element containing the rest of string. If the limit parameter is negative, all components except the last -limit are returned. If the limit parameter is zero, then this is treated as 1.
+-- @return table
+function string.explode(s, sep, limit)
+    if not sep or sep == "" then return false end
```
Comment (Afforess) [inline stdlib/string.lua]: if `s` or `sep` are required arguments, the standard practice is to throw an error. For example:

```
require 'stdlib/core'
...
function string.explode(s, sep, limit)
    fail_if_missing(s, "missing string argument")
    fail_if_missing(sep, "missing separator character")
```

If you want to be thorough, you could add tests to verify the failure conditions properly throw an error.

Comment (credomane) [inline stdlib/string.lua]: Yeah I need to do that, sorry. Was working on my Config module. I just copy-n-pasta'd this in did a few tests and PR'd it after getting upset with string.split. >_>

Comment (Afforess) [inline stdlib/string.lua]: No worries, I just thought the `return false` would be confusing for a method that reports it returns a table.

File: spec/string_spec.lua [hunk 2]
```diff
@@ -76,7 +76,38 @@ describe('String Spec', function()
...
+            assert.has.errors(function() string.explode("","ok") end) --string arg cannot be empty
+            assert.has.errors(function() string.explode("Test","") end) --sep arg cannot be empty
+            assert.has.errors(function() string.explode(1,"good") end) -- string arg must be a string
+            assert.has.errors(function() string.explode("1",1) end) -- sep arg must be a string
+            assert.has.errors(function() string.explode("Test","string","") end) --limit must be number
+        end)
+
+        it('test limit arg', function()
+            local s = "This is a test"
+            assert.same({}, s:explode(" ", 0-4))
```
Comment (credomane) [inline spec/string_spec.lua]: `0-4`? What I get for only double checking and not triple-checking.

**Comment by credomane** [comment]
The revised string.split makes this obsolete unless someone *really* wants the limit feature. Closing this PR.

**Pull request closed without merging**